api/proto/teleport/decision/v1alpha1/ssh_identity.proto

```diff
@@ -16,9 +16,174 @@ syntax = "proto3";
 
 package teleport.decision.v1alpha1;
 
+import "google/protobuf/timestamp.proto";
+import "teleport/decision/v1alpha1/tls_identity.proto";
+import "teleport/trait/v1/trait.proto";
+
 option go_package = "github.com/gravitational/teleport/api/gen/proto/go/teleport/decision/v1alpha1;decisionpb";
 
 // SSHIdentity is the identity used for SSH connections.
 message SSHIdentity {
-  // TBD
+  // --- common identity fields ---
```

Comment:
Does it make sense to split these categories into submessages?

Reply:
I lean against, mostly because keeping them as one is much more consistent with how teleport seems to prefer to work currently.  Teleport really likes not caring about whether something is a host or a user for most operations, and keeping it all flat makes that easier.  I could definitely see the argument that maybe it would be better if we did treat them more as conceptually separate, but I'm trying to as much as possible not change anything substantive during this refactoring, and creating a stronger conceptual split between host and user identities seems like it will likely have a long tail of behavioral changes and/or bugs.